support/cas-server-support-person-directory/src/main/java/org/apereo/cas/config/CasPersonDirectoryConfiguration.java

```diff
@@ -267,9 +267,13 @@ public IPersonAttributeDao attributeRepository() {
         final List<IPersonAttributeDao> list = new ArrayList<>();
         casProperties.getAuthn().getAttributeRepository().getScript()
             .forEach(Unchecked.consumer(script -> {
-                final ScriptEnginePersonAttributeDao dao = new ScriptEnginePersonAttributeDao();
-                final String scriptFile = IOUtils.toString(script.getLocation().getInputStream(), StandardCharsets.UTF_8);
-                dao.setScriptFile(scriptFile);
+                final String scriptContents = IOUtils.toString(script.getLocation().getInputStream(), StandardCharsets.UTF_8);
+                final String engineName = script.getEngineName() == null ?
+                        ScriptEnginePersonAttributeDao.getScriptEngineName(script.getLocation().getFilename()) : script.getEngineName();
+                if (engineName == null) {
```

Comment:
Might this check be better placed inside the DAO itself, if it is not there already?